### DISCUSSION TITLE
Lateral direction motion modes (Dutch-roll, Spiral Mode, Roll) in JSBSim

### DISCUSSION THREAD

**Discussion by vrottolo** [JSBSim-Team/jsbsim#1086] [Q&A]
Hello,

I'm interested in exciting and visualizing lateral directional motion modes such as Dutch-roll, Spiral Mode, and Roll using JSBSim.

Could anyone please guide me on how to excite and visualize these lateral modes? Any pointers or resources would be greatly appreciated.

Thanks!

**Comment by seanmcleod** [comment]
Do you understand the flight dynamics of the Dutch roll etc and just need to understand how to move the controls in JSBSim in order to excite them?

**Reply by vrottolo** [reply]
Yes, exactly! I want to excite them and visualize the responses over time. I am using Python to interface with JSBSim.

For example, I would like to try reproducing the following technique, as described in "DTIC ADA319979: Volume II. Flying Qualities Phase. Chapter 8: Dynamics":

Spiral Mode Flight Test Technique:
"Trim the aircraft for hands-off flight, paying particular attention to lateral control and centering the ball. Roll into a 20° bank in one direction, release the controls, and measure the bank angle after 20 seconds. Repeat the maneuver in a bank to the opposite side."

Roll Mode Flight Test Technique:
"Trim the aircraft for hands-off flight. Roll into a 45° bank in one direction and stabilize the aircraft. Abruptly apply a small step aileron input and hold it throughout a 90° change in bank angle. The size of the step aileron input should be sufficiently small to allow the aircraft to achieve a steady-state roll rate prior to the 90° bank angle change, but sufficiently large to measure the roll rate with the instrumentation system onboard the aircraft."

I wonder if it's possible.

**Comment by seanmcleod** [comment]
Some techniques are going to be easier than others. For example for the Dutch roll

https://en.wikipedia.org/wiki/Dutch_roll
> The Dutch roll mode can be excited by any use of [aileron](https://en.wikipedia.org/wiki/Aileron) or [rudder](https://en.wikipedia.org/wiki/Rudder), but for flight test purposes it is usually excited with a rudder singlet (a short sharp motion of the rudder to a specified angle, and then back to the centered position) or doublet (a pair of such motions in opposite directions). Some larger aircraft are better excited with aileron inputs. Periods can range from a few seconds for [light aircraft](https://en.wikipedia.org/wiki/Light-sport_aircraft) to a minute or more for [airliners](https://en.wikipedia.org/wiki/Airliner).

So in this case you could simply use JSBSim's trim routine to trim the aircraft at some specified altitude and speed and then simply input a singlet or doublet of either the rudder or ailerons and then graph the output.

And if you wanted to/needed to you could edit the relevant stability derivatives in the aircraft's FDM in order to demonstrate how the Dutch roll dynamics change.

For some of the other techniques, e.g. you'll need additional logic over and above simply inputting a singlet, doublet etc.

Take a look at the following discussion for an example where I used Python to implement a rudder-kick maneuver that someone was asking about - https://github.com/JSBSim-Team/jsbsim/discussions/519
